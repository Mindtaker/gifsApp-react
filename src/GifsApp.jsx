import { useState } from 'react';

import { AddCategory, GifGrid } from './components';

export const GifsApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (category) => {
    const newCategory = category.toLowerCase();
    if (categories.some((cat) => cat.toLowerCase() === newCategory)) return;

    setCategories([category, ...categories]);
  };

  return (
    <>
      <h1>Gifs App</h1>
      <AddCategory onNewCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
