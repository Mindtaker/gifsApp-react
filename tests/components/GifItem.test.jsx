import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('GifItem component', () => {
  const url = 'https://www.test.com/test.jpg';
  const title = 'test';

  test('should match the snapshot', () => {
    const { container } = render(<GifItem url={url} title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('should show the image with the given url', () => {
    render(<GifItem url={url} title={title} />);
    
    const imgElement = screen.getByRole('img');
    expect(imgElement.src).toBe(url);
    expect(imgElement.alt).toBe(title);
  });
  
  test('should show the title of the gif', () => {
    render(<GifItem url={url} title={title} />);
    
    expect(screen.getByText(title)).toBeTruthy();
  });
});
