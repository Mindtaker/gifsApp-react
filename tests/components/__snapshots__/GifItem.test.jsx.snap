// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="https://www.test.com/test.jpg"
    />
    <p>
      test
    </p>
  </div>
</div>
`;
